src/FieldWarning/Assets/Units/Component/Damage/Damage.cs

```diff
@@ -0,0 +1,102 @@
+﻿/**
+ * Copyright (c) 2017-present, PFW Contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Diagnostics;
+
+namespace PFW.Units.Component.Damage
+{
+    /// <summary>
+    /// The base class Damage, on which all damage classes are constructed
+    /// This class represents a damage dealt in an update, for one-time instant damage (e.g. KEDamage or HeatDamage), the class will be instantiated once
+    /// For continuous damage (e.g. FireDamage), the class will be instantiated once and the CalculateDamage will be called at every update (controlled by another script inherenting MonoBehaviour)
+    /// </summary>
+    internal abstract class Damage
```

Comment:
Personally I would do this with composition instead of inheritance. Meaning I'd make this thing or its contents a member variables of the various damage types. By using inheritance you get to avoid some code duplication but the code becomes very hard to read or change, now your HEDamage class has a bunch of (**!public!**) variables that you need to open a separate file to see, some of them maybe you don't even care about (Era), and God help you with the mayhem if you want to ever change something in the base Damage class in the future.